### PR TITLE
Show app configuration in sidebar for non-admin users

### LINKED ISSUE #1997: Cannot Access App Configuration without Admin Privileges after 2021.12
**iOS device model, version and app version**

Model Name: iPhone XS
Software Version: 15.1 (19B74)
App version: 2021.11.1

**Home Assistant Core Version**
core-2021.12.0

**Describe the bug**
Without admin role on a user that user is not able to edit their app configuration.
I know that App Configuration was moved into Configuration menu (#1993), though I don't believe it is expected that users without the admin role should no be able to access this configuration. 

**To Reproduce**

**Expected behavior**
I would expect users without the admin role should be able to either have the app configuration menu item back or show them just that entry in the configuration menu rather than hiding the configuration and making it impossible to access.

### LINKED ISSUE #10894: Non-Administrator Users dont have access to the Companion App Settings
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Since the union of the settings in 2021.12 users which are not assigned to admin role can no longer go to the app configuration since it is hidden in the not accessible settings screen now.  

### Describe the behavior you expected

As a non admin user I still want to configure my cpmpanion app settings.

### Steps to reproduce the issue

1. Log in as a non admin user
2. No settings are shown in the sidebar

### What version of Home Assistant Core has the issue?

core-2021.12.1

### What was the last working version of Home Assistant Core?

core-2021.11.x

### In which browser are you experiencing the issue with?

Android Web View

### Which operating system are you using to run this browser?

Android 12

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Revert #10733

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/iOS/issues/1997 closes home-assistant/core#61711
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Shouldn't we only add it in the sidebar if the user doesn't have access to the config panel?

**Comment by ludeeus** [comment]
Why would we do that?
If admin, go to config, if not go to sidebar? that's confusing for everyone.

File: src/components/ha-sidebar.ts [hunk 1]
```diff
@@ -548,6 +561,34 @@ class HaSidebar extends LitElement {
     </a>`;
   }
 
+  private _renderExternalConfiguration() {
+    return html`${this._externalConfig && this._externalConfig.hasSettingsScreen
```
Comment (balloob) [inline src/components/ha-sidebar.ts:565]: Wwe should only render this if the user is non-admin.

**Comment by TomBrien** [comment]
> Why would we do that?
> 
> If admin, go to config, if not go to sidebar? that's confusing for everyone.

I have to agree, it seems very confusing to have the menu in different places depending on user rights

File: src/components/ha-sidebar.ts [hunk 2]
```diff
@@ -548,6 +557,43 @@ class HaSidebar extends LitElement {
     </a>`;
   }
 
+  private _renderExternalConfiguration() {
+    return html`${!this.hass.user?.is_admin &&
```
Comment (bramkragten) [inline src/components/ha-sidebar.ts:561]: We should check if we show the config panel, as a admin user could also have disabled the config integration (yes, edge case)

Comment (balloob) [reply src/components/ha-sidebar.ts:561]: it's not possible because `config` is a dependency of `frontend`

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
We are going to revert this change for non-admin users in a hot fix and will work on a more proper solution in a normal release.

**Pull request merged**